src/test/java/bio/terra/service/tabulardata/google/BigQueryPdaoTest.java

```diff
@@ -403,17 +405,65 @@ private void softDeleteRows(BigQueryProject bq,
+
+            storage.create(participantBlob, readFile("ingest-test-participant.json"));
+            storage.create(sampleBlob, readFile("ingest-test-sample.json"));
+            storage.create(fileBlob, readFile("ingest-test-file.json"));
+
+            // Ingest staged data into the new dataset.
+            IngestRequestModel ingestRequest = new IngestRequestModel()
+                .format(IngestRequestModel.FormatEnum.JSON);
+
+            String datasetId = dataset.getId().toString();
+            connectedOperations.ingestTableSuccess(datasetId,
+                ingestRequest.table("participant").path(gsPath(participantBlob)));
+            connectedOperations.ingestTableSuccess(datasetId,
+                ingestRequest.table("sample").path(gsPath(sampleBlob)));
+            connectedOperations.ingestTableSuccess(datasetId,
+                ingestRequest.table("file").path(gsPath(fileBlob)));
 
-        DatasetTable participantTable = getTable(dataset, "participant");
-        DatasetTable sampleTable = getTable(dataset, "sample");
-        DatasetTable fileTable = getTable(dataset, "file");
-        List<DatasetTable> tables = Arrays.asList(participantTable, sampleTable, fileTable);
+            // Create a full-view snapshot!
+            DatasetSummaryModel datasetSummary = DatasetJsonConversion.datasetSummaryModelFromDatasetSummary(
+                dataset.getDatasetSummary());
+            SnapshotSummaryModel snapshotSummary = connectedOperations.createSnapshot(datasetSummary,
+                "snapshot-fullviews-test-snapshot.json", "");
+            SnapshotModel snapshot = connectedOperations.getSnapshot(snapshotSummary.getId());
 
-        bigQueryPdao.createSnaphotTableFromLiveViews(bigQueryProject, tables, datasetBqDatasetName);
+            BigQueryProject bigQueryProject = TestUtils.bigQueryProjectForDatasetName(
+                datasetDao, dataLocationService, dataset.getName());
+            Assert.assertThat(snapshot.getTables().size(), is(equalTo(3)));
+            List<String> participantIds = queryForIds(snapshot.getName(), "participant", bigQueryProject);
+            List<String> sampleIds = queryForIds(snapshot.getName(), "sample", bigQueryProject);
+            List<String> fileIds = queryForIds(snapshot.getName(), "file", bigQueryProject);
+
+            Assert.assertThat(participantIds, containsInAnyOrder(
+                "participant_1", "participant_2", "participant_3", "participant_4", "participant_5"));
+            Assert.assertThat(sampleIds, containsInAnyOrder(
+                "sample1", "sample2", "sample3", "sample4", "sample5", "sample6", "sample7"));
+            Assert.assertThat(fileIds, is(equalTo(Collections.singletonList("file1"))));
+        } finally {
+            storage.delete(participantBlob.getBlobId(), sampleBlob.getBlobId(), fileBlob.getBlobId());
+            bigQueryPdao.deleteDataset(dataset);
+            datasetDao.delete(dataset.getId());
```

Comment:
This is the extra cleanup logic I originally wanted to add. Without it, the `tearDown` method tries to delete the resource profile & encounters a FK integrity violation

Reply:
maybe add this as a comment in the code? I'm curious as to how we were getting around the FK integrity violation and whether we could get into the same state in other tests.

Reply:
Happy to add a comment somewhere, but not sure of the best place (this diff makes the test case look the same as all the other tests in the file). I could add the comment to the `tearDown` line that triggers the FK violation, but I'm not sure who would read it... Alt I could add it to every test case for max copy-paste-ability. WDYT?

Reply:
Decided to do both, more comments don't hurt anybody